ckan/views/__init__.py

```diff
@@ -43,27 +50,142 @@ def set_cors_headers_for_response(response: Response) -> Response:
     return response
 
 
+# ETag's are very useful since it allows conditional freshness checks
+# without bandwidth costs by the browser/cdn.
+# Since html pages from ckan are always generated on the fly it
+# can't use last modified and content length our Nginx does it
+def set_etag_and_fast_304_response_if_unchanged(response: Response) -> Response:
```

Comment:
This middleware would be better as a separate optional plugin.

As is it assumes that bandwidth is expensive and memory/processing time on the web server are cheap. This won't be true for all users. It also adds a significant amount of code to the handling of every request.